platforms/access-bridge/src/services/stripe-service.ts

```diff
@@ -0,0 +1,60 @@
+import Stripe from 'stripe';
+
+import { STRIPE_SECRET } from '../app-config.js';
+
+export type StripeProduct = Stripe.Product & {
+  prices: Stripe.Price[];
+};
+
+/**
+ * Service class responsible for interacting with the Stripe API to fetch products.
+ */
+export class StripeService {
+  private stripe: Stripe;
+
+  constructor() {
+    this.stripe = new Stripe(STRIPE_SECRET, {
+      apiVersion: '2024-06-20',
+    });
+  }
+
+  /**
+   * Retrieves Stripe products with prices and filters them based on the provided productIds.
+   * @param productIds The array of product IDs to filter the products.
+   * @returns A Promise resolving to an array of filtered ProductsWithMetadata objects.
+   */
+  async getProductsWithPrices(productIds: string[]): Promise<StripeProduct[]> {
```

Comment:
Maybe this method should return the already-mapped results that we plan to use like:
```
{
  plan_id: 123456
  price: $55
  recurrence: daily
  description: ....
}
```
This is related to the suggestion above that this implement a generic `PaymentService` interface and decouples us from stripe at higher levels, keeping stripe specific logic encapsulated in the stripe service.

Reply:
(the json above is just a sample. I don't know what we actually plan to pick out and use to show to viewers)

Reply:
It is exactly what it does. It retrieves the mapped Stripe product. We are retrieving the entire Stripe product as we don't wanna use our own typings in case of Stripe changes.